File: code-samples/voice/call-forwarding.java
import com.ringcentral.*;
import com.ringcentral.definitions.*;

public class Create_ForwardingNumber {
    public static void main(String[] args) {
        try {
            create_forwarding_number();
        } catch (RestException | IOException e) {
            e.printStackTrace();
        }
    }

    public static void create_forwarding_number() throws RestException, IOException {
        RestClient rc = new RestClient("client_id", "client_secret", "server_url");
        rc.authorize("username", "extension_number", "password");

        var parameters = new CreateForwardingNumberRequest();
        parameters.phoneNumber = "11235557890";
        parameters.type = "Other";
        parameters.label = "My ATT number";

        var response =  rc.restapi().account().extension().forwardingnumber().post(parameters);

        System.out.println("Forwarding number created.");
        System.out.println(response.id);
    }
}

File: code-samples/voice/call-forwarding.cs
using System;
using System.Threading.Tasks;
using RingCentral;

namespace Create_ForwardingNumber
{
  class Program
  {
    static RestClient restClient;
    static async Task Main(string[] args)
    {
      restClient = new RestClient(
      Environment.GetEnvironmentVariable("RC_APP_CLIENT_ID"),
      Environment.GetEnvironmentVariable("RC_APP_CLIENT_SECRET"),
      Environment.GetEnvironmentVariable("RC_SERVER_URL"));
      await restClient.Authorize(Environment.GetEnvironmentVariable("RC_USER_JWT"));

      await create_forwarding_number();
    }
    static private async Task create_forwarding_number()
    {
      var parameters = new CreateForwardingNumberRequest();
      parameters.phoneNumber = "11235557890";
      parameters.type = "Other";
      parameters.label = "My ATT number";

      var response = await restClient.Restapi().Account().Extension().ForwardingNumber().Post(parameters);

      Console.WriteLine("Forwarding number created.");
      Console.WriteLine(response.id);
    }
  }
}
